frontend/src/components/activity/ActivityResponsibles.vue

```diff
@@ -71,21 +73,54 @@ export default {
       return this.activity.camp().campCollaborations()
     },
   },
-  async mounted() {
-    await this.activityResponsibles._meta.load
-    this.oldSelectedCampCollaborations = [...this.currentCampCollaborationIRIs]
-    this.selectedCampCollaborations = [...this.currentCampCollaborationIRIs]
+  watch: {
+    activity: {
+      async handler(newActivity, oldActivity) {
+        // Activity changed; reset SelectedCampCollaboration-Arrays
+        if (oldActivity?._meta.self != newActivity._meta.self) {
```

Comment:
I can add an additional `?`, but I think the code is already correct.
- if `oldActivity` is undefined or null, then `oldActivity?._meta.self` is undefined
- if `oldActivity` is defined, then it must contain a `_meta` object (https://github.com/ecamp/hal-json-vuex/blob/next/src/interfaces/ResourceInterface.ts)

In our codebase we rely on this a lot (129 hits for `_meta.self` vs 2 hits for `_meta?.self`).

Reply:
> - if `oldActivity` is undefined or null, then `oldActivity?._meta.self` is undefined

No, if oldActivity is undefined or null, then oldActivity?._meta.self will produce the console error "oldActivity?._meta is undefined", or, as Chrome puts it, [Cannot read properties of undefined](https://sentry.io/organizations/ecamp/issues/3538682667/?project=5602507). oldActivity?._meta returns the value undefined in that case, and next it will try to evaluate undefined.self, leading to the above error. If you ever use the optional chaining operator, you have to use it until the end of the call chain.
The operator does not return some magical object which you can do anything with. That magical object exists in ecamp, but only hal-json-vuex produces it. And if oldActivity is null or undefined, hal-json-vuex doesn't even come into play here.

> In our codebase we rely on this a lot (129 hits for `_meta.self` vs 2 hits for `_meta?.self`).

That is simply because in most cases, the foo in foo._meta.self cannot be null or undefined, and so we don't have to use question mark operator at all. You should regex search for usages of `\?\._meta\?\.self` vs. `[^?]\._meta\.self`

Reply:
> If you ever use the optional chaining operator, you have to use it until the end of the call chain.

This statement is not correct.

Documentation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
<img width="397" alt="image" src="https://user-images.githubusercontent.com/449555/201466250-0756c037-7fa6-4112-8dd5-6f2a0ef8e55f.png">

Example code: https://playcode.io/1009594
(only the last one throws, which is not expected because meta is a mandatory property of our resources)

Reply:
Wow, nevermind! I did not know that, thanks!

Reply:
No worries 😃 learning every day

Thanks for merging